reactive/README.md

```diff
@@ -0,0 +1,310 @@
+# Reactive addon
+
+## Summary
+
+This PR adds a new addon called `reactive`, which is an alternative to `hooks` for managing state. It's main purpose is to decouple the rendering phase from state updates to ensure optimal rendering performance by default, even for complex apps. It comes with some DX improvements, such as that dependencies are tracked automatically and don't have to be specified by hand for effects.
+
+## Basic Example
+
+```jsx
+import { signal } from 'preact/reactive';
+
+// Won't re-render despite state being updated. The effect will
+// trigger only when `count.value` changes, not when the component
+// re-renders.
+function App() {
+	const [count, setCount] = signal(0);
+
+	// Look, no dependency args
+	effect(() => {
+		console.log(`count: ${count.value}`);
+	});
+
+	return <NeverReRendered count={count} />;
+}
+
+// Will only render once during mounting, will not re-render
+// when state is updated in <App />
+function NeverReRendered({ count }) {
+	return <Inner count={count} />;
+}
+
+// Whenever the button is clicked, only the <Inner /> component will rerender
+function Inner({ count }) {
```

Comment:
Should `setCount` be passed down as well? If not, how does `Inner` have access to it?

```suggestion
	return <NeverReRendered count={count} setCount={setCount} />;
}

// Will only render once during mounting, will not re-render
// when state is updated in <App />
function NeverReRendered({ count, setCount }) {
	return <Inner count={count} setCount={setCount} />;
}

// Whenever the button is clicked, only the <Inner /> component will rerender
function Inner({ count, setCount }) {
```

Intriguing PR btw!